python/dgl/graphbolt/minibatch.py

```diff
@@ -231,6 +231,19 @@ def blocks(self):
             self.sampled_subgraphs[0].sampled_csc, Dict
         )
 
+        # casts to minimum dtype in-place and returns self.
+        def cast_to_minimum_dtype(v: CSCFormatBase):
```

Comment:
Should we do it in the compact stage?

Reply:
We can consider doing it in the compact stage as an optimization to fuse the type casting into the compact operation to save runtime.